Associative-Containers/exercise2.cpp

```diff
@@ -17,25 +17,18 @@ enum class ErrorCode
     None
 };
 
-bool BadFormat (const std::string &input_)
-{
-    if (input_[0] == '+' && '/' && '*' && '%'){return true;}
-    return false;
-}
+
 
 bool DivideByZero (const std::string &input_)
 {
-    for(size_t i = 0;i < input_.size();i++)
-        { 
-            
-            if (input_[i] == '/' && input_[i+2] == '0')
-            {
-                
-                return true;
-            
-            }
-        }   
-        return false;
+   std::stringstream input_stream(input_);
+   double x,y;
+   char op;
+   if(!(input_stream>>x)){return false;}
+   if(!(input_stream>>op)){return false;}
+   if(!(input_stream>>y)){return false;}
```

Comment:
```suggestion
   if(!(input_stream>>x) || !(input_stream>>op) || !(input_stream>>y))
   {
    return false;
   }
```

There is really important programming rule called DRY, which means Don't Repeat Yourself. It says you should avoid code-duplication. 

In Polish: https://infotraining.bitbucket.io/cpp-tdd/refactoring.html#duplikacja-kodu

Actually, operator>> is [associative](https://en.wikipedia.org/wiki/Associative_property#Definition) (po polsku działanie nazywamy łącznym, tak jak dodawanie tzn. `(a + b) + c = a + (b + c)` dla dowolnych liczb) so you can simplified it more like this:
```suggestion
   if(!(input_stream >> x >> op >> y))
   {
    return false;
   }
```